src/clients/Client.cc

```diff
@@ -847,18 +849,17 @@ void Client::handleAdaptedBodyProducerAborted()
     if (abortOnBadEntry("entry went bad while waiting for the now-aborted adapted body"))
         return;
 
+    adaptedReplyAborted = true;
     Must(adaptedBodySource != nullptr);
     if (!adaptedBodySource->exhausted()) {
         debugs(11,5, "waiting to consume the remainder of the aborted adapted body");
         return; // resumeBodyStorage() should eventually consume the rest
     }
```

Comment:
I do not like that we are duplicating this check, but I do not yet know whether we can avoid that duplication. I am aware that the check is needed here to protect us from a premature handledEarlyAdaptationAbort() call if there are still Store-unwritten adapted body bytes in adaptedBodySource buffer.